tyk-data-plane/README.md

```diff
@@ -0,0 +1,172 @@
+`tyk-redis-master.tyk.svc.cluster.local:6379` (Tyk needs the name including the port) 
+
+### Pump Configurations
+
+To enable Pump, set `global.components.pump` to true
+
+| Pump       | Configuration      |
+| ---------- | ------------------ | 
+| Prometheus Pump (Default)  | Set `tyk-pump.pump.backend='prometheus'`, and add connection details for prometheus under `tyk-pump.pump.prometheusPump`. |
+| Mongo Pump | Set `tyk-pump.pump.backend='mongo'`, and add connection details for mongo under `global.mongo`. |
+| SQL Pump   | Set `tyk-pump.pump.backend='postgres'`, and add connection details for postgres under `global.postgres`. |
+| Uptime Pump | Set `tyk-pump.pump.uptimePumpBackend` to `'mongo'` or `'postgres'` or `''` |
+| Other Pumps    | Set `tyk-pump.pump.backend=''` if you want to configure other pump types  |
+
+#### Prometheus Pump
+Set `tyk-pump.pump.backend='prometheus'`, and add connection details for prometheus under `tyk-pump.pump.prometheusPump`. 
+
+#### Mongo pump
+If you are using the MongoDB pumps in the tyk-data-plane installation you will require MongoDB installed for that as well.
+
+To install Mongo you can use these rather excellent charts provided by Bitnami:
+
+    helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk
+
+(follow notes from the installation output to get connection details and update them in `values.yaml` file)
+
+NOTE: [Here is](https://tyk.io/docs/planning-for-production/database-settings/) list of supported MongoDB versions. Please make sure you are installing mongo helm chart that matches these version.
+
+*Important Note regarding MongoDB:* This helm chart enables the PodDisruptionBudget for MongoDB with an arbiter replica-count of 1. If you intend to perform system maintenance on the node where the MongoDB pod is running and this maintenance requires for the node to be drained, this action will be prevented due the replica count being 1. Increase the replica count in the helm chart deployment to a minimum of 2 to remedy this issue.
+
+Add following under the `global` section in `values.yaml`:
+
+     # Set mongo connection details if you want to configure mongo pump.     
+     mongo:
+        # The mongoURL value will allow you to set your MongoDB address.
+        # Default value: mongodb://mongo.{{ .Release.Namespace }}.svc.cluster.local:27017/tyk_analytics
+        # mongoURL: mongodb://mongo.tyk.svc.cluster.local:27017/tyk_analytics
+        # If your MongoDB has a password you can add the username and password to the url
+        # mongoURL: mongodb://root:pass@tyk-mongo-mongodb.tyk.svc.cluster.local:27017/tyk_analytics?authSource=admin
+        mongoURL: <MongoDB address>
+
+        # Enables SSL for MongoDB connection. MongoDB instance will have to support that.
+        # Default value: false
+        # useSSL: false
```

Comment:
similar to https://github.com/TykTechnologies/tyk-charts/pull/24#discussion_r1144516052